src/main/java/io/snyk/jenkins/SnykStepBuilder.java

```diff
@@ -264,7 +264,7 @@ public FormValidation doCheckSnykTokenId(@AncestorInPath Item item, @QueryParame
         }
       }
       if (fixEmptyAndTrim(value) == null) {
-        return FormValidation.error("Snyk API token is required.");
+        return FormValidation.warningWithMarkup("A Snyk API token is required. If you do not provide credentials, make sure to provide a <code>SNYK_TOKEN</code> build environment variable.");
```

Comment:
Could we reliably detect if it's set and not print it?

Reply:
No, as build environment variables are only known at build time (they're dynamic).